pkg/datastore/postgres/repositories/checked_headers_repository_test.go

```diff
@@ -30,313 +32,378 @@ import (
-		It("marks headers with matching block number as unchecked", func() {
-			blockNumberOne := rand.Int63()
-			blockNumberTwo := blockNumberOne + 1
-			blockNumberThree := blockNumberOne + 2
-			fakeHeaderOne := fakes.GetFakeHeader(blockNumberOne)
-			fakeHeaderTwo := fakes.GetFakeHeader(blockNumberTwo)
-			fakeHeaderThree := fakes.GetFakeHeader(blockNumberThree)
-			headerRepository := repositories.NewHeaderRepository(db)
-			// insert three headers with incrementing block number
-			headerIdOne, insertHeaderOneErr := headerRepository.CreateOrUpdateHeader(fakeHeaderOne)
-			Expect(insertHeaderOneErr).NotTo(HaveOccurred())
-			headerIdTwo, insertHeaderTwoErr := headerRepository.CreateOrUpdateHeader(fakeHeaderTwo)
-			Expect(insertHeaderTwoErr).NotTo(HaveOccurred())
-			headerIdThree, insertHeaderThreeErr := headerRepository.CreateOrUpdateHeader(fakeHeaderThree)
-			Expect(insertHeaderThreeErr).NotTo(HaveOccurred())
-			// mark all headers checked
-			markHeaderOneCheckedErr := repo.MarkHeaderChecked(headerIdOne)
-			Expect(markHeaderOneCheckedErr).NotTo(HaveOccurred())
-			markHeaderTwoCheckedErr := repo.MarkHeaderChecked(headerIdTwo)
-			Expect(markHeaderTwoCheckedErr).NotTo(HaveOccurred())
-			markHeaderThreeCheckedErr := repo.MarkHeaderChecked(headerIdThree)
-			Expect(markHeaderThreeCheckedErr).NotTo(HaveOccurred())
-
-			// mark header from blockNumberTwo unchecked
-			err := repo.MarkSingleHeaderUnchecked(blockNumberTwo)
-
-			Expect(err).NotTo(HaveOccurred())
-			var headerOneCheckCount, headerTwoCheckCount, headerThreeCheckCount int
-			getHeaderOneErr := db.Get(&headerOneCheckCount, `SELECT check_count FROM public.headers WHERE id = $1`, headerIdOne)
-			Expect(getHeaderOneErr).NotTo(HaveOccurred())
-			Expect(headerOneCheckCount).To(Equal(1))
-			getHeaderTwoErr := db.Get(&headerTwoCheckCount, `SELECT check_count FROM public.headers WHERE id = $1`, headerIdTwo)
-			Expect(getHeaderTwoErr).NotTo(HaveOccurred())
-			Expect(headerTwoCheckCount).To(BeZero())
-			getHeaderThreeErr := db.Get(&headerThreeCheckCount, `SELECT check_count FROM public.headers WHERE id = $1`, headerIdThree)
-			Expect(getHeaderThreeErr).NotTo(HaveOccurred())
-			Expect(headerThreeCheckCount).To(Equal(1))
-		})
-	})
+			It("increments check count on update", func() {
+				headerRepository := repositories.NewHeaderRepository(db)
+				headerID, headerErr := headerRepository.CreateOrUpdateHeader(fakes.FakeHeader)
+				Expect(headerErr).NotTo(HaveOccurred())
 
-	Describe("UncheckedHeaders", func() {
-		var (
-			headerRepository datastore.HeaderRepository
-			firstBlock,
-			secondBlock,
-			thirdBlock,
-			lastBlock,
-			secondHeaderID,
-			thirdHeaderID int64
-			blockNumbers        []int64
-			headerIDs           []int64
-			err                 error
-			uncheckedCheckCount = int64(1)
-			recheckCheckCount   = int64(2)
-		)
+				Expect(repo.MarkHeaderChecked(headerID)).To(Succeed())
+				Expect(repo.MarkHeaderChecked(headerID)).To(Succeed())
```

Comment:
nbd but might we want a stronger assert than `Succeed()`? Seems like we could verify that the val is incremented

Reply:
lol lost in the diff, I see that the assert I'm describing is below 😅

Reply:
So I ended up changing a lot of this test because I learned to features of Gomega that I don't see a lot of in our tests.  It's documented here: 

https://onsi.github.io/gomega/#handling-errors

The short version is if you return value, err from a function (as you often do from Go funcs) and you use To(Equal(blah)) that ALSO validates that any other values in the tuple are nil. So you don't have to constantly check for error.

In addition above `Expect(repo.MarkHeaderChecked(headerID)).To(Succeed())` - requires all tuples to be nil returned from `MarkHeaderChecked`. This works best for functions like `MarkHeaderChecked` that only return an error.

I really like the first bit (we could remove a lot of explicit error checking, as it's redundant). The second I like for setup/action steps like MarkHeaderChecked is here, but I don't love it, and I'm really using it more like a traditional assert where something has gone horribly wrong if it fails.

Anyway I'm highlighting this because most of our tests aren't written this way, but I'll probably start migrating tests over to use less assertions when appropriate unless there's an objection.

Reply:
👍 absolutely love it